File: spring-data-microservice-transaction-sample/order-service/src/main/java/sample/order/domain/model/Order.java
package sample.order.domain.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Table("orders")
public class Order {

  @Id
  public final String orderId;
  public final int customerId;
  public final long timestamp;

  public Order(String orderId, int customerId, long timestamp) {
    this.orderId = orderId;
    this.customerId = customerId;
    this.timestamp = timestamp;
  }
}

File: scalardb-dotnet-samples/scalardb-cluster-linq-sample/Models/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using ScalarDB.Client.DataAnnotations;

namespace ScalarDbClusterLinqSample.Models;

[Table("sample.orders")]
public class Order
{
    [SecondaryIndex]
    [Column("order_id", Order = 0)]
    public string Id { get; set; } = "";
    
    [PartitionKey]
    [Column("customer_id", Order = 1)]
    public int CustomerId { get; set; }

    [ClusteringKey]
    [Column("timestamp", Order = 2)]
    public long Timestamp { get; set; }
}
